scipy/signal/_short_time_fft.py

```diff
@@ -0,0 +1,1659 @@
+"""Short-time Fourier Transform Module.
+
+Implementation Notes (as of 2023-04)
+------------------------------------
+* MyPy version 1.1.1 does not seem to support decorated property methods
+  properly. Hence, applying ``@property`` to methods decorated with `@cache``
+  (as tried with the ``lower_border_end`` method) causes a mypy error when
+  accessing it as an index (e.g., ``SFT.lower_border_end[0]``).
+* MyPy version 1.1.1 does not seem to support ``NDArray[complex]`` (only
+  ``NDArray[np.complex128]``). Hence the methods `stft()`, `stft_detrend()`,
+  `istft()`, and  `_ifft_func()` need to use `NDArray` array as a parameter
+  instead of `NDArray[complex]`.
+* The entry ``.. currentmodule:: scipy.signal.ShortTimeFFT`` is required
+  in docstrings to ensure that `stft` and `istft` generate HTML links to
+  methods and not the legacy functions. The side effect is that `ShortTimeFFT`
+  does not work anymore. It is not known how to do this file-wide instead of
+  method-wide.
```

Comment:
I am not sure I see the issue here, but I don't think we need to do this in other places.

Reply:
Writing `` `stft` `` in a docstring generates a link to the legacy implementation. Using ``.. currentmodule:: scipy.signal.ShortTimeFFT`` circumvents this.